### PR TITLE
Add doc links to liballoc crate page

### PR DESCRIPTION
**Pull request by czipperz**

### CODE REVIEW

File: src/liballoc/lib.rs [hunk 1]
```diff
@@ -49,6 +49,12 @@
 //!
 //! The [`alloc`](alloc/index.html) module defines the low-level interface to the
 //! default global allocator. It is not compatible with the libc allocator API.
+//!
+//! [`Arc`]: sync/index.html
+//! [`Box`]: boxed/index.html
+//! [`Cell`]: ../core/cell/index.html
```
Comment (czipperz) [inline src/liballoc/lib.rs]: I'm not sure this is the right location.

**Comment by gagan0723** [comment]
ping from triage @QuietMisdreavus any updates on this?

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**